C4G/BLIS
# Update documentation, devcontainer, and language files

### BLIS 3.8 pre-release improvements

#### `.devcontainer` updates

* Added packages to support building the documentation site inside the devcontainer
* Added the `shellcheck` extension to the VS Code config for linting shell scripts
* Modifications to the `vscode` user to grant additional permissions to `www-data`-owned files

#### Tooling updates

* `bin/build-blis-release.sh`: This shell script will create zipped release and upgrade packages for BLIS, based on the current master branch (not the files in the directory you are running from.) It is similar to `Build-BLIS.ps1`, but works slightly better as the Powershell version does not work well outside of Windows, nor does it generate an upgrade package.
* `bin/php-cs-fixer`: Changed this to a shell script that invokes the PHP script rather than a symlink to avoid problems with Windows
* `bin/update-lang.php`: Added this script which is able to generate a `.php` language file from a language `.xml` file. This can be used to manually update the PHP versions the locales (en, fr, default) after updating the XML files by hand.

#### Language File Update & Developer Tooling

* There were significant updates to each locale file (en, default) after manually auditing them and adding missing strings by hand. **The French language files were not updated because I am not able to verify the correct French strings.**

* These changes are only intended to apply to _new labs._ They are included in the default language files and the language files used by the standalone BLIS package. Upgrades are not intended to be affected.

### Testing

#### Standalone

**[[[ Current Standalone test zip ]]](https://gtvault-my.sharepoint.com/:u:/g/personal/mrysavy3_gatech_edu/EW17LfNDVPxFrw92KXjgz68BiruGjR0LU1O84n1LxfSUjA?e=D5QZte)**

Download this zip file and **test it as a standalone BLIS package** by running BLIS.exe.

#### Upgrade

**[[[ Current Upgrade test zip ]]](https://gtvault-my.sharepoint.com/:u:/g/personal/mrysavy3_gatech_edu/EY0oAScLNdxDni3lleefsOgB7AOUNZESGkV9ZU4HrtfciA?e=b6Nfsi)**

Download this zip file and **test it as an upgrade BLIS package.**

To test this scenario:

1. Download [BLIS 3.72](http://blis.cc.gatech.edu/files/C4G_BLIS_Version_3.72.zip?) from the BLIS website and run it as normal.
2. Make some changes that you would recognize: add a new lab, add some patient data, add some tests, etc.
3. Stop BLIS (close Firefox)
4. Download the upgrade zip above. Unzip it and copy all the files in the `BLIS-Upgrade/` folder into your BLIS 3.72 folder.
  - When asked if you want to replace or merge files - yes!
5. Start BLIS
6. Log in as an admin, superadmin, or country director.
7. You should see a prompt to upgrade to BLIS 3.8: click that and ensure it is successful.
8. Search the site for the changes you made above to ensure they are still there.

#### Testing Scenarios

1. Make a backup in BLIS 3.72, upgrade to BLIS 3.8 (see above), and restore the backup
2. Record patient data as you would expect to and ensure all pages work
